model_api/cpp/models/src/segmentation_model.cpp

```diff
@@ -32,8 +32,80 @@
 #include "models/input_data.h"
 #include "models/results.h"
 
+namespace {
+cv::Mat create_hard_prediction_from_soft_prediction(const cv::Mat& soft_prediction, float soft_threshold, int blur_strength) {
+    cv::Mat soft_prediction_blurred;
+    cv::blur(soft_prediction, soft_prediction_blurred, cv::Size{blur_strength, blur_strength});
+    assert(soft_prediction_blurred.channels() > 1);
+    cv::Mat hard_prediction{cv::Size{soft_prediction_blurred.rows, soft_prediction_blurred.cols}, CV_8UC1};
+    for (int i = 0; i < soft_prediction_blurred.rows; ++i) {
+        for (int j = 0; j < soft_prediction_blurred.cols; ++j) {
+            float max_prob = -std::numeric_limits<float>::infinity();
+            uint8_t max_id = 0;
+            for (int c = 0; c < soft_prediction_blurred.channels(); ++c) {
+                float prob = ((float*)soft_prediction_blurred.data)[
```

Comment:
Here Mat.ptr(i) could be used to reduce the indexing expression

Reply:
Updated